rust/template/differential_datalog/Cargo.toml

```diff
@@ -13,6 +13,7 @@ c_api = []
 [dependencies]
 #differential-dataflow = "0.11.0"
 differential-dataflow = { git = "https://github.com/ddlog-dev/differential-dataflow", branch = "ddlog-2" }
+dogsdogsdogs = { git = "https://github.com/ddlog-dev/differential-dataflow", branch = "ddlog-2" }
```

Comment:
Unrelated to this pr, d3 looks very promising for a lot of the joins that ddlog does, are there plans to use it elsewhere within the runtime?

Reply:
I have support for d3 implemented in a branch, which has now bit rotted and will need to be refreshed.  Unfortunately, while it works and the performance is basically the same as with regular DD operators, I never got much improvement in terms of memory footprint out of it.  The main reason is that, while you don't need to arrange every prefix of the rule anymore, you need to construct additional arrangements of individual relations in the rule.  In addition, d3 does not work with recursion, and I don't think I figured out how to express antijoins with it.  But this is all very use case-specific, so I agree it would be nice to support it.

Reply:
> d3 does not work with recursion

Actually this is no longer true.  DD now contains differentiation and integration operators that can be used to implement delta queries inside recursive fragments of DD: [blog](https://github.com/frankmcsherry/blog/blob/master/posts/2020-02-15.md)